pkg/plugin/group/scaler_test.go

```diff
@@ -13,7 +13,7 @@ func TestScaleUp(t *testing.T) {
 	defer ctrl.Finish()
 
 	scaled := mock_group.NewMockScaled(ctrl)
-	scaler := NewScalingGroup(scaled, 3, 1*time.Millisecond)
+	scaler := NewScalingGroup(scaled, 3, 1*time.Millisecond, 0)
```

Comment:
Can we add a new test or test this with `maxParallel = 1`?   Maybe change the case to create more than one instance.  The test should pass fine since we are effectively serializing the CreateOne calls.